base/src/types/mod.rs

```diff
@@ -520,6 +545,12 @@ pub enum Type<Id, T = ArcType<Id>> {
     Opaque,
     /// A builtin type
     Builtin(BuiltinType),
+    /// Universally quantified types
+    Forall(
+        #[cfg_attr(feature = "serde_derive", serde(state))] Vec<Generic<Id>>,
+        #[cfg_attr(feature = "serde_derive", serde(state))] T,
+        #[cfg_attr(feature = "serde_derive", serde(state))] Option<Vec<T>>,
```

Comment:
What is the `Option<Vec<T>>` here?

Reply:
I try to explain it in https://github.com/gluon-lang/gluon/pull/320#discussion_r148456145